### PR TITLE
Fix/upload collection of files

### PR DESCRIPTION
**Pull request by IsaMic**
|Q              |A  |
|---------------|---|
|Bug fix?       |Y|
|New feature?   |N|	
|BC breaks?     |N|	
|Deprecations?  |N|	
|Fixed tickets  |N|	

Debug if we have a collection of fields with a file field in a Form

### CODE REVIEW

File: Helper/Api/ApiService.php [hunk 1]
```diff
@@ -45,40 +46,78 @@ public function __construct(LoggerInterface $logger, \Twig_Environment $twig, Ur
...
+    /**
+     *
+     * @param array<string,mixed> $body
+     * @param string $apiName
+     * @param FileBag<array>|array<string,mixed> $files
+     * @return array<string,mixed>
+     */
+    private function treatFiles(array $body, string $apiName, $files): array
+    {
+        $body_field = $body;
```
Comment (dameert) [inline Helper/Api/ApiService.php:71]: $bodyF
```suggestion
        $bodyField = $body;
```

File: Helper/Api/ApiService.php [hunk 2]
```diff
@@ -45,40 +46,78 @@ public function __construct(LoggerInterface $logger, \Twig_Environment $twig, Ur
...
+        /** @var string $fieldKey */
+        foreach ($files as $fieldKey => $fileField) {
+            if (is_array($fileField)) {
```
Comment (dameert) [inline Helper/Api/ApiService.php:74]:
```suggestion
            if (\is_array($fileField)) {
```

File: Helper/Api/ApiService.php (hunk 1 above)
Comment (dameert) [reply Helper/Api/ApiService.php:71]: Did you test without $body_field = $body; ?

Just using $body directly. At first glance this seems like a possibility?

**Review by dameert — CHANGES_REQUESTED** [review]

File: Helper/Api/ApiService.php (hunk 1 above)
Comment (IsaMic) [reply Helper/Api/ApiService.php]: Yes, $body_field must be deleted and use $body directly.

File: Helper/Api/ApiService.php [hunk 3]
```diff
@@ -45,40 +46,77 @@ public function __construct(LoggerInterface $logger, \Twig_Environment $twig, Ur
...
+    private function treatFiles(array $body, string $apiName, $files): array
+    {
+        /** @var string $fieldKey */
+        foreach ($files as $fieldKey => $fileField) {
+            if (\is_array($fileField)) {
+                /** @var string $pos */
+                foreach ($fileField as $pos => $collectionOfFields) {
+                    /** @var string $fileKey */
+                    foreach ($collectionOfFields as $fileKey => $file) {
+                        if (is_array($file)) {
```
Comment (dameert) [inline Helper/Api/ApiService.php:78]:
```suggestion
                        if (\is_array($file)) {
```
;-)

File: Helper/Api/ApiService.php [hunk 4]
```diff
@@ -45,40 +46,77 @@ public function __construct(LoggerInterface $logger, \Twig_Environment $twig, Ur
...
+     *
+     * @param string $apiName
+     * @param UploadedFile $file
+     * @throws \Exception
+     * @return array<string, mixed>
+     */
+    private function createContentFileHashField(string $apiName, UploadedFile $file): array
+    {
+        $response = $this->uploadFile($apiName, $file, $file->getClientOriginalName());
+        if (!$response['uploaded'] || !isset($response[EmsFields::CONTENT_FILE_HASH_FIELD_])) {
```
Comment (dameert) [inline Helper/Api/ApiService.php:106]:
```suggestion
        if (! $response['uploaded'] || ! \isset($response[EmsFields::CONTENT_FILE_HASH_FIELD_])) {
```

**Review by dameert — CHANGES_REQUESTED** [review]

File: Helper/Api/ApiService.php (hunk 4 above)
Comment (IsaMic) [reply Helper/Api/ApiService.php:106]: not possible on isset()

**Review by dameert — APPROVED** [review]

**Pull request merged**